### ISSUE TITLE
Android – Request – “Next step” is partly visible in the WS with IOU if device font size is larger

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#33950]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.21-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Log in
3. Make a device font size larger than the default
4. Navigate to the WS with IOU
5. Open IOU page

## Expected Result:
“Next step” is fully visible in the WS with IOU if device font size larger than the default .

## Actual Result:
“Next step” is partly visible in the WS with IOU if device font size larger than the default .

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![image](https://github.com/Expensify/App/assets/93399543/0410ef29-f0c7-466c-b690-8f68552bc751)

https://github.com/Expensify/App/assets/93399543/e9d9be6e-b1f4-4c6c-88c3-3fdd96c149d1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When we increase device font size, the Next Step text is partially visible because it's really big.

### What is the root cause of that problem?
In the Next Step font size dynamically increases as we increase the system font size. We actually use a `textMicroBold` style for the Next Step text and the max font size for it is 17.
https://github.com/Expensify/App/blob/d5f5856b610cc39d5b5f2b054dab67bb74d66922/src/components/MoneyReportHeaderStatusBar.tsx#L27

https://github.com/Expensify/App/blob/d5f5856b610cc39d5b5f2b054dab67bb74d66922/src/styles/index.ts#L375-L379

But we use RN Text component to render the Next Step, not our custom Text.

### What changes do you think we should make in order to solve the problem?

Replace RN Text with our custom Text component

**Comment by johncschuster** [comment]
@c3024 can you review the proposal above? Is it sufficient, or should we keep an eye out for more?

**Comment by c3024** [comment]
Have issues with building Android app. Will review ASAP.

**Comment by johncschuster** [comment]
@c3024 how's your android build coming along? Are you still having trouble with it?

**Comment by c3024** [comment]
@bernhardoj 's [proposal here](https://github.com/Expensify/App/issues/33950#issuecomment-1877071216) looks good to me.

🎀 👀 🎀 C+ Reviewed

**Comment by c3024** [comment]
Yes. I've not pulled the latest main when I reviewed it. Thanks.

The issue can be closed.
